src/index.rs

```diff
@@ -356,142 +389,123 @@ impl ResourceIndex {
         let path_buf = CanonicalPathBuf::canonicalize(path)?;
         let path = path_buf.as_canonical_path();
 
-        return match fs::metadata(path) {
-            Err(_) => {
-                return Err(ArklibError::Path(
-                    "Couldn't to retrieve file metadata".into(),
-                ));
-            }
+        match fs::metadata(path) {
+            Err(_) => Err(ArklibError::Path(
+                "Couldn't to retrieve file metadata".into(),
+            )),
             Ok(metadata) => match scan_entry(path, metadata) {
-                Err(_) => {
-                    return Err(ArklibError::Path(
-                        "The path points to a directory or empty file".into(),
-                    ));
+                Err(_) => Err(ArklibError::Path(
+                    "The path points to a directory or empty file".into(),
+                )),
+                Ok(entry) => {
+                    let result = IndexUpdate::added(path_buf.clone(), entry.id);
+                    self.insert_entry(path_buf, entry);
+                    Ok(result)
```

Comment:
We can also do this to reduce the indentation levels and improve readability.

```suggestion
        let metadata = fs::metadata(path).map_err(|_| {
            ArklibError::Path("Couldn't retrieve file metadata".into())
        })?;

        let entry = scan_entry(path, metadata).map_err(|_| {
            ArklibError::Path(
                "The path points to a directory or empty file".into(),
            )
        })?;

        let result = IndexUpdate::added(path_buf.clone(), entry.id);
        self.insert_entry(path_buf, entry);
        Ok(result)
```

Reply:
This looks really great!